lighthouse-core/audits/apple-touch-icon.js

```diff
@@ -0,0 +1,92 @@
+ */
+'use strict';
+
+const Audit = require('./audit.js');
+const URL = require('../lib/url-shim');
+const i18n = require('../lib/i18n/i18n.js');
+
+/**
+ * @fileoverview Audits if a page's web app has an icon link for iOS installability.
+ */
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that tells the user that their site contains a vaild touch icon. This descriptive title is shown when the page contains a valid iOS touch icon. */
+  title: 'Provides a valid apple-touch-icon',
+  /** Title of a Lighthouse audit that tells the user that their site contains a vaild touch icon. This descriptive title is shown when the page does not contain a valid iOS touch icon. */
+  failureTitle: 'Does not provide a valid apple-touch-icon',
+  /** Description of a Lighthouse audit that tells the user what having a valid apple-touch-icon does. This is displayed after a user expands the section to see more. No character length limits. 'Learn More' becomes link text to additional documentation. */
+  description: 'In order to have a custom icon on iOS once ' +
+  'installed, a site must have a valid apple-touch-icon. ' +
+  '[Learn More](https://developers.google.com/web/fundamentals/design-and-ux/browser-customization/).',
+  /** Warning that HTML attribute `apple-touch-icon-precomposed` should not be used in favor of `apple-touch-icon`.  "apple-touch-icon-precomposed" and "apple-touch-icon" are HTML attributes and should not be translated. */
+  precomposedWarning: '`apple-touch-icon-precomposed` is out of date, ' +
+  '`apple-touch-icon` is preferred.',
+  /** Explanatory message stating that there was a failure in an audit caused by the page's `apple-touch-icon` having an invalide `href` attribute. `apple-touch-icon` and `href` are HTML tag values and should not be translated. */
+  explanation: '`apple-touch-icon`\'s `href` attribute is not valid',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class IosPwaIcon extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'apple-touch-icon',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['LinkElements'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    let explanation;
+    const appleTouchIcons = artifacts.LinkElements.filter(
+      el => {
+        if (el.rel !== 'apple-touch-icon' && el.rel !== 'apple-touch-icon-precomposed'
+          || !el.href) {
+          return false;
+        }
+        // Check that the href is valid
+        try {
+          new URL(el.href);
+          return true;
+        } catch (e) {}
+        // Check if the href needs the base url to be valid
```

Comment:
you can just combine these checks (the base url is ignored if the first argument is already absolute)

Reply:
It doesn't seeeeeem like that works....

Reply:
got it!  finalUrl was undefined, and was silently throwing and caught by the audit.